### DISCUSSION TITLE
Invalid Host Defined Options

### DISCUSSION THREAD

**Discussion by v4lentin1879** [withcatai/node-llama-cpp#224] [Q&A] [answered]
I'm trying to include node-llama-cpp in a visual studio code extension using webpack.
The application works fine in debugging mode. However, as soon as I package it and install the vsix file, when running the application, I get the "TypeError: Invalid host defined options". The error occurs on the dynamic import of the node-llama-cpp.

I would be super happy if anyone could help me here.. @giladgd 
```
        //workaround to import esm module
        const importPath = "node-llama-cpp";
        const {getLlama, LlamaChatSession, LlamaGrammar} = await Function(`return import("${importPath}")`)();
```

**Comment by giladgd** [comment, marked as answer]
Closing for now as we talked about it in [this thread](https://github.com/withcatai/node-llama-cpp/discussions/81#discussioncomment-9558282)
